src/main/java/hudson/remoting/Request.java

```diff
@@ -238,7 +238,10 @@ public RSP get() throws InterruptedException, ExecutionException {
                             if (isCancelled()) {
                                 throw new CancellationException();
                             }
-                            Request.this.wait(); // wait until the response arrives
+                            if (channel.isInClosed()) {
+                                throw new ExecutionException(new RequestAbortedException(null));
```

Comment:
Nope that's what this exception pattern means (see non-async call method above)

:sadpanda: